pkg/server/ca/upstream_client.go

```diff
@@ -69,13 +73,13 @@ func (u *UpstreamClient) Close() error {
 // open stream to the UpstreamAuthority plugin to receive and append X.509 root
 // updates to the bundle. The stream remains open until another call to
 // MintX509CA happens or the client is closed.
-func (u *UpstreamClient) MintX509CA(ctx context.Context, csr []byte, ttl time.Duration) (_ []*x509.Certificate, err error) {
+func (u *UpstreamClient) MintX509CA(ctx context.Context, csr []byte, ttl time.Duration, validateX509CA ValidateX509CAFunc) (_ []*x509.Certificate, err error) {
 	u.mintX509CAMtx.Lock()
 	defer u.mintX509CAMtx.Unlock()
 
 	firstResultCh := make(chan mintX509CAResult, 1)
 	u.mintX509CAStream.Start(func(streamCtx context.Context) {
-		u.runMintX509CAStream(streamCtx, csr, ttl, firstResultCh)
+		u.runMintX509CAStream(streamCtx, csr, ttl, validateX509CA, firstResultCh)
```

Comment:
may we validate that function is not nil?

Reply:
I think you are touching on a more philosophical question :) There are lots of required yet nil-able parameters that we don't validate (e.g. contexts). If we add an if-check, it's going to be impossible to actually unit-test that condition since the parameter is provided by the manager unconditionally... I think my inclination would be to not check at runtime.